ingest-app/test/cmr/ingest/api/subscriptions_test.clj

```diff
@@ -1,31 +1,83 @@
 (ns cmr.ingest.api.subscriptions-test
   (:require
+   [cheshire.core :as json]
    [clojure.string :as string]
    [clojure.test :refer :all]
    [cmr.ingest.api.core :as api-core]
-   [cmr.ingest.api.subscriptions :as subscriptions]))
-
-(deftest ingest-subscription-test
-  (testing "creating a subscription"
-    (let [subscription {}]
-      (testing "with native-id provided uses the native-id"
-        (with-redefs-fn {#'subscriptions/perform-subscription-ingest (constantly nil)
-                         #'subscriptions/common-ingest-checks (constantly nil)
-                         #'api-core/body->concept! (constantly {:native-id "tmp"
-                                                                :metadata " {\"Name\": \"some name\"}"})
-                         #'subscriptions/check-subscription-ingest-permission (fn [request-context concept provider-id]
-                                                                                (is (= "given-native-id" (:native-id concept))))}
-          #(subscriptions/ingest-subscription "test-provider" "given-native-id" subscription)))
-
-      (testing "with native-id not provided generates a native-id"
-        (with-redefs-fn {#'subscriptions/perform-subscription-ingest (constantly nil)
-                         #'subscriptions/common-ingest-checks (constantly nil)
-                         #'api-core/body->concept! (constantly {:native-id "tmp"
-                                                                :metadata " {\"Name\": \"some name\"}"})
-                         #'subscriptions/check-subscription-ingest-permission
-                         (fn [request-context concept provider-id]
-                           (is (string/starts-with? (:native-id concept) "some_name")))}
-          #(subscriptions/ingest-subscription "test-provider" nil subscription))))))
+   [cmr.ingest.api.subscriptions :as subscriptions]
+   [cmr.transmit.metadata-db :as mdb]
+   [ring.mock.request :as mock]))
+
+(deftest create-subscription-test
+  (testing "with native-id not provided generates a native-id"
+    (let [request (-> (mock/request :post
+                                    "/PROV1/subscriptions"
+                                    {:content-type "application/vnd.nasa.cmr.umm+json;version=1.0"})
+                      (mock/json-body {:Name "no native id"
+                                       :SubscriberId "post-user"
+                                       :EmailAddress "someone@speedy.com"
+                                       :CollectionConceptId "C1200000018-PROV1"
+                                       :Query "polygon=-18,-78,-13,-74,-16,-73,-22,-77,-18,-78"}))]
+      (with-redefs-fn {#'subscriptions/common-ingest-checks (constantly nil)
```

Comment:
Not a fan of using stub functions for functional tests. This setup is too rigid and hard to maintain. These probably can be replaced with unit tests. The integration tests should cover the slack without functional tests. I really want to delete these tests, but I will leave it to you if you see value in doing this.

Reply:
I dislike this as well, but the structure of the code makes increasing code coverage impossible unless mocks are used.